libs/utils/src/hmpb/ballot_adjudication_reasons.ts

```diff
@@ -1,46 +1,50 @@
 import {
   AdjudicationReason,
   AdjudicationReasonInfo,
-  Contest,
   ContestOption,
   ContestOptionId,
   Contests,
   MarkStatus,
+  WriteInAreaStatus,
 } from '@votingworks/types';
 import { throwIllegalValue } from '@votingworks/basics';
 import { allContestOptions } from './all_contest_options';
 
 export interface Options {
-  optionMarkStatus: (option: ContestOption) => MarkStatus;
+  optionStatus: (option: ContestOption) => {
+    markStatus: MarkStatus;
+    writeInAreaStatus: WriteInAreaStatus;
+  };
 }
 
 /**
  * Enumerates all the reasons a series of contests might need adjudication.
- * Callers must provide a function that can get the mark status for any contest
- * option in the contests given.
+ * Callers must provide a function that can get the mark status and write-in
+ * area status for any contest option in the contests given.
  */
 export function* ballotAdjudicationReasons(
   contests: Contests,
-  { optionMarkStatus }: Options
+  { optionStatus }: Options
```

Comment:
It looks like the `optionStatus` parameter is only used in one place. Can we just move the logic of that function into this function? It's kind of confusing to go back and forth to figure out what is determining the option status, and it seems like we don't actually need the flexibility of having it as a parameter.

Reply:
The way that this logic is split between the adapter and `utils` is confusingly specific, and I opted to keep it in case there was an equally specific justification it.

I think that it should just be moved from `utils` into `ballot-interpreter`.

Reply:
Agreed